src/zarr_jpeg/zarr_jpeg.py

```diff
@@ -10,25 +10,38 @@ class jpeg(Codec):
     Parameters
     ----------
     quality : int
-        Compression level. 
+        Compression level.
     """
 
     codec_id = "jpeg"
 
-    def __init__(self, quality=100):
+    def __init__(self, quality=100, axis_reduction='collapse'):
         self.quality = quality
+        self.axis_reduction = axis_reduction
         assert (
             self.quality > 0 and self.quality <= 100 and isinstance(self.quality, int)
         )
         super().__init__()
 
     def encode(self, buf):
         bufa = ensure_ndarray(buf)
-        assert bufa.ndim == 2 or bufa.ndim == 3
-        if bufa.ndim == 3:
-            tiled = bufa.reshape(bufa.shape[0] * bufa.shape[1], -1)
-        else:
+        if isinstance(self.axis_reduction, str) and self.axis_reduction == 'collapse':
+            if bufa.ndim >= 3:
+                # all but the last dimension are collapsed into first dimension, followed by last dimension
+                self.axis_reduction = [[dim for dim in range(bufa.ndim - 1)]]
+                self.axis_reduction.append([bufa.ndim - 1])
+            else:
```

Comment:
looks like this will allow 1D arrays through, which will raise an exception from the jpeg codec. I think we need to probably error if the input is not 2D or greater.

Reply:
I've added an assert bufa.ndim >= 2

Reply:
Actually, there is an edge case that we could support.  A 1-dimensional buffer could be reshaped with `axis_reduction = [[], [0]]` to a 2-dimensional buffer before being handed to the codec.  That is, if we remove the `assert bufa.ndim >= 2` statement then the existing code would happily change a 1-dimensional buffer with shape `(256,)` to a 2-dimensional buffer with shape `(1, 256)` when given `axis_reduction = [[], [0]]`.

However, giving the user that kind of power might not be all that useful, and the confusion and the failure to give errors on 1-dimensional arrays could actually be counterproductive.  What do you think?

Reply:
Given how jpeg works, I think nobody should be using jpeg for compressing 1D data. If someone really wants to do that, they should reshape their data directly... but if there's some forseeable use case I could be swayed in the other direction?

Reply:
I personally have no need for jpeg compression of 1-dimensional arrays.  We might simply keep `assert bufa.ndim >= 2` unless and until someone asks to have it removed.  Removing it later is not likely to break anyone's code -- unless, what, if there is a `try:` block that is expecting 1-dimensional arrays to throw exceptions that are then caught and subject to specific processing in an `except:` block.